sarracen/interpolate.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+
+
+def interpolate2D(data, x, y, target, kernel, xmin, ymin, pixwidthx, pixwidthy, pixcountx, pixcounty):
+    """
+    Interpolates particle data in a SarracenDataFrame across two directional axes to a 2D
+    grid of pixels.
+
+    :param data: The particle data, in a SarracenDataFrame.
+    :param x: The column label of the x-directional axis.
+    :param y: The column label of the y-directional axis.
+    :param target: The column label of the target smoothing data.
+    :param kernel: The kernel to use for smoothing the target data.
+    :param xmin: The starting x-coordinate (in particle data space).
+    :param ymin: The starting y-coordinate (in particle data space).
+    :param pixwidthx: The width that each pixel represents in particle data space.
+    :param pixwidthy: The height that each pixel represents in particle data space
+    :param pixcountx: The number of pixels in the output image in the x-direction.
+    :param pixcounty: The number of pixels in the output image in the y-direction.
+    :return: The output image, in a 2-dimensional numpy array.
+    """
+    image = np.zeros((pixcounty, pixcountx))
+
+    # iterate through all pixels
+    for i, particle in data.iterrows():
+        # dimensionless weight
+        # w_i = m_i / (rho_i * (h_i) ** 2)
+        weight = particle['m'] / (particle['rho'] * particle['h'] ** 2)
+
+        # skip particles with 0 weight
+        if weight <= 0: continue
+
+        # kernel radius scaled by the particle's 'h' value
+        radkern = kernel.radkernel * particle['h']
+        term = weight * particle[target]
+        hi1 = 1 / particle['h']
+        hi21 = hi1 ** 2
+
+        part_x = particle[x]
+        part_y = particle[y]
+
+        # determine the min/max x&y coordinates affected by this particle
+        ipixmin = int(np.rint((part_x - radkern - xmin)/np.abs(pixwidthx)))
+        jpixmin = int(np.rint((part_y - radkern - ymin)/np.abs(pixwidthy)))
+        ipixmax = int(np.rint((part_x + radkern - xmin)/np.abs(pixwidthx)))
+        jpixmax = int(np.rint((part_y + radkern - ymin)/np.abs(pixwidthy)))
```

Comment:
Can skip the np.abs() on the pixwidth here. See above comment about value error checks related to this.